api/files/permissions.py

```diff
@@ -6,6 +6,9 @@
 class CheckedOutOrAdmin(permissions.BasePermission):
     def has_object_permission(self, request, view, obj):
         assert isinstance(obj, FileNode), 'obj must be a FileNode, got {}'.format(obj)
+        if request.method in permissions.SAFE_METHODS:
+            return True
```

Comment:
Is this the case? Probably is (thinking my way through now). Even if the file is checked out, you should be able to read it, so should be fine. That's what you were thinking?